src/Spice86/ViewModels/RegisterViewModel.cs

```diff
@@ -0,0 +1,142 @@
+namespace Spice86.ViewModels;
+
+using CommunityToolkit.Mvvm.ComponentModel;
+
+using Spice86.Core.Emulator.CPU;
+
+using System;
+
+/// <summary>
+/// View model for a CPU register value.
+/// </summary>
+public partial class RegisterViewModel : ObservableObject {
+    private readonly State _state;
+    private readonly Func<State, uint> _valueGetter;
+    private uint _previousValue;
+
+    /// <summary>
+    /// Gets the name of the register.
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// Gets the current value of the register.
+    /// </summary>
+    public uint Value { get; private set; }
+
+    /// <summary>
+    /// Gets the hexadecimal representation of the register value.
+    /// </summary>
+    public string HexValue => $"{Value:X4}";
+
+    /// <summary>
+    /// Gets a value indicating whether the register value has changed since the last update.
+    /// </summary>
+    public bool HasChanged { get; private set; }
+
+    /// <summary>
+    /// Gets a value indicating whether the low byte (bits 0-7) has changed.
+    /// </summary>
+    public bool LowByteChanged { get; private set; }
+
+    /// <summary>
+    /// Gets a value indicating whether the high byte (bits 8-15) has changed.
+    /// </summary>
+    public bool HighByteChanged { get; private set; }
+
+    /// <summary>
+    /// Gets a value indicating whether the upper word (bits 16-31) has changed.
+    /// </summary>
+    public bool UpperWordChanged { get; private set; }
+
+    /// <summary>
+    /// Gets a value indicating whether the lower word (bits 0-15) has changed.
+    /// </summary>
+    public bool LowerWordChanged { get; private set; }
+
+    /// <summary>
+    /// Gets the low byte (bits 0-7) in hexadecimal.
+    /// </summary>
+    public string LowByteHex => $"{Value & 0xFF:X2}";
```

Comment:
We have a (badly named) class "ConvertUtils" which already has some of those things.
Wouldn't it make sense to put code that is cutting words and dwords in a shared place?

Reply:
I'd rather keep them here where they're used. "external" changes to this functionality can really mess up the registers view.
It's also easier to see whats going on, versus having to come up with good names for new functions in the ConvertUtils.